resolwe_bio/processes/reads_processing/bbduk.yml

```diff
@@ -11,221 +11,363 @@
     expression-engine: jinja
     executor:
       docker:
-        image: resolwebio/legacy:1.0.0
-  data_name: '{{ reads.fastq.0.file|default("?") }}'
-  version: 0.0.6
+        image: resolwebio/rnaseq:3.0.0
+    resources:
+      cores: 10
+  data_name: '{{ reads|sample_name|default("?") }}'
+  version: 1.0.0
   type: data:reads:fastq:single:bbduk
   category: analyses
   flow_collection: sample
   persistence: CACHED
   description: >
-    BBduk is capable of quality-trimming and filtering, adapter-trimming, contaminant-filtering via
-    kmer matching, sequence masking, GC-filtering, length filtering, entropy-filtering, format
-    conversion, histogram generation, subsampling, quality-score recalibration, kmer cardinality
-    estimation, and various other operations in a single pass.
+    BBDuk combines the most common data-quality-related trimming, filtering, and masking operations
+    into a single high-performance tool. It is capable of quality-trimming and filtering,
+    adapter-trimming, contaminant-filtering via kmer matching, sequence masking, GC-filtering,
+    length filtering, entropy-filtering, format conversion, histogram generation, subsampling,
+    quality-score recalibration, kmer cardinality estimation, and various other operations in a
+    single pass. See
+    [the official website](https://jgi.doe.gov/data-and-tools/bbtools/bb-tools-user-guide/bbduk-guide/)
+    for more information.
   input:
     - name: reads
       label: Reads
       type: data:reads:fastq:single
-    - name: adapters
-      label: Adapters
+    - name: show_advanced
+      label: Show advanced parameters
+      type: basic:boolean
+      default: false
+    - name: reference
+      label: Reference
```

Comment:
Missing description of the input field. Its not obvious what the `reference` is. Maybe try to explain that here a user should provide a multi-sequence .fasta file with adapters

Reply:
Good idea! I like the concept of a reference. It encompasses several types of sequences such as adapters, contaminants, and degenerate sequences like primers. I added additional description as you suggested. Note, however, that description of a `group` is not displayed on the front-end. I therefore added the description to the first input in this group.